ghost/admin/app/services/migrate.js

```diff
@@ -0,0 +1,121 @@
+import Service, {inject as service} from '@ember/service';
+import {SignJWT} from 'jose';
+import {tracked} from '@glimmer/tracking';
+
+export default class MigrateService extends Service {
+    @service ajax;
+    @service router;
+    @service ghostPaths;
+    @service settings;
+
+    migrateUrl = 'http://localhost:3010';
+    migrateRouteRoot = '#/migrate';
+
+    @tracked migrateWindowOpen = false;
+    @tracked siteData = null;
+    @tracked previousRoute = null;
+    @tracked isIframeTransition = false;
+
+    get apiUrl() {
+        const origin = window.location.origin;
+        const subdir = this.ghostPaths.subdir;
+        const rootURL = this.router.rootURL;
+        let url = this.ghostPaths.url.join(origin, subdir, rootURL);
+        url = url.replace(/\/$/, ''); // Strips the trailing slash
+        return url;
+    }
+
+    async apiToken() {
+        // TODO: Getting the token can be improved
+        const ghostIntegrationsUrl = this.ghostPaths.url.api('integrations') + '?include=api_keys';
+        return this.ajax.request(ghostIntegrationsUrl).then(async (response) => {
+            const ssmIntegration = response.integrations.find(r => r.slug === 'self-serve-migration');
+
+            const key = ssmIntegration.api_keys[0].secret;
+            const [id, secret] = key.split(':');
```

Comment:
we are expecting that this version will be released along with the [key formatting fix](https://github.com/TryGhost/Ghost/pull/16508), so there will be nothing to update in the future. sweeet 👌